cluster/cluster.go

```diff
@@ -226,20 +209,34 @@ type ClusterAlerts interface {
 	EraseAlert(resource api.ResourceType, alertID int64) error
 }
 
-// Cluster is the API that a cluster provider will implement.
-type Cluster interface {
+// ClusterClient is an interface specific to the exported functions accessable by the
+// client of OpenStorage.
+type ClusterClient interface {
 	// Inspect the node given a UUID.
 	Inspect(string) (api.Node, error)
 
-	// AddEventListener adds an event listener and exposes cluster events.
-	AddEventListener(ClusterListener) error
-
 	// Enumerate lists all the nodes in the cluster.
 	Enumerate() (api.Cluster, error)
 
 	// SetSize sets the maximum number of nodes in a cluster.
 	SetSize(size int) error
 
+	// Remove node(s) from the cluster permanently.
+	Remove(nodes []api.Node, forceRemove bool) error
+
+	ClusterStatus
+	ClusterAlerts
+	ClusterData
+
+	// This may be removed from being exposed over REST: See #383
```

Comment:
I made a comment on #383 and have created a JIRA 5154 to track decision on this as well.